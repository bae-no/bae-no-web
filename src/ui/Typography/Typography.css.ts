import { style } from "@vanilla-extract/css";
import { recipe, RecipeVariants } from "@vanilla-extract/recipes";

const fontSize = {
  typo1: { fontSize: "1.875rem", lineHeight: "2.5rem" },
  typo2: { fontSize: "1.625rem", lineHeight: "2.1875rem" },
  typo3: { fontSize: "1.375rem", lineHeight: "1.8125rem" },
  typo4: { fontSize: "1.125rem", lineHeight: "1.5625rem" },
  typo5: { fontSize: "1rem", lineHeight: "1.3125rem" },
  typo6: { fontSize: "0.9375rem", lineHeight: "1.25rem" },
  typo7: { fontSize: "0.8125rem", lineHeight: "1.125rem" },
};

const fontWeight = {
  bold: { fontWeight: "700" },
  semiBold: { fontWeight: "600" },
  regular: { fontWeight: "400" },
};

const base = style({
  overflowWrap: `break-word`,
  fontFamily:
    "Pretendard, -apple-system, BlinkMacSystemFont, system-ui, Roboto, Helvetica Neue, Segoe UI, Apple SD Gothic Neo, Noto Sans KR, Malgun Gothic, sans-serif",
});

export const fontStyle = recipe({
  base,
  variants: { fontSize, fontWeight },
  defaultVariants: { fontSize: "typo5", fontWeight: "regular" },
});

export type FontStyle = RecipeVariants<typeof fontStyle>;
