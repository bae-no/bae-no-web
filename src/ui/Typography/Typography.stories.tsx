import { ComponentStory, ComponentMeta } from "@storybook/react";

import { colors } from "../tokens/color";

import Typography from "./Typography";

export default {
  argTypes: {
    color: { control: "select", options: Object.keys(colors) },
  },
  component: Typography,
  title: "UI/Typography",
} as ComponentMeta<typeof Typography>;

const Template: ComponentStory<typeof Typography> = (args) => (
  <Typography {...args} />
);

export const Default = Template.bind({});
Default.args = {
  children: "Default",
};
