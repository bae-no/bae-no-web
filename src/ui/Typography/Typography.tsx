import { forwardRef, HtmlHTMLAttributes } from "react";

import { Box } from "../Box";
import { Sprinkles } from "../sprinkles.css";

import { FontStyle, fontStyle } from "./Typography.css";

type TypographyProps = Pick<
  HtmlHTMLAttributes<HTMLParagraphElement>,
  "aria-label" | "children"
> &
  Pick<Sprinkles, "color"> &
  FontStyle & {
    as?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "p" | "label" | "span";
  };

const Typography = forwardRef(
  ({ children, as = "p", color, fontSize, ...rest }: TypographyProps, ref) => (
    <Box
      {...rest}
      as={as}
      className={fontStyle({
        fontSize,
      })}
      color={color}
      ref={ref}
    >
      {children}
    </Box>
  )
);

export default Typography;
