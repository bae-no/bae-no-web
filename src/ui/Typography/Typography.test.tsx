import { render, screen } from "@testing-library/react";
import Typography from "./Typography";

describe("UI Typography Component", () => {
  it("match snapshot", () => {
    const { container } = render(
      <Typography fontSize="typo2" fontWeight="bold" />
    );

    expect(container).toMatchSnapshot();
  });

  it("render text", () => {
    render(<Typography>test text</Typography>);

    expect(screen.getByText("test text")).toBeInTheDocument();
  });
});
