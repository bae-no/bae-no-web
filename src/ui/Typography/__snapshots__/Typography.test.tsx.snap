// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Typography Component match snapshot 1`] = `
<div>
  <p
    class="sprinkles_display_flex_mobile__1vna7gb3 sprinkles_flexDirection_column_mobile__1vna7gbi reset_base__4op0mk0 Typography_base__1rfncg10 Typography_fontStyle_fontSize_typo2__1rfncg12 Typography_fontStyle_fontWeight_bold__1rfncg18"
  />
</div>
`;
